dora/core/common/src/main/java/alluxio/membership/WorkerServiceEntity.java

```diff
@@ -57,26 +61,37 @@ public WorkerServiceEntity() {}
 
   /**
    * CTOR  for WorkerServiceEntity with given WorkerNetAddress.
+   *
+   * @param identity
    * @param addr
    */
-  public WorkerServiceEntity(WorkerNetAddress addr) {
-    super(HashUtils.hashAsStringMD5(addr.dumpMainInfo()));
+  public WorkerServiceEntity(WorkerIdentity identity, WorkerNetAddress addr) {
```

Comment:
Wondering if a new constructor can be added to take the WorkerServiceEntity as an input parameter